x-pack/test/functional/services/ml/test_resources.ts

```diff
@@ -514,17 +516,23 @@ export function MachineLearningTestResourcesProvider(
       await this.assertSavedObjectExistsById(id, SavedObjectType.DASHBOARD);
     },
 
-    async deleteMlSavedObjectByJobId(jobId: string, jobType: JobType) {
+    async deleteMlSavedObjectByJobId(jobId: string, jobType: JobType, space?: string) {
       const savedObjectId = `${jobType}-${jobId}`;
-      await this.deleteSavedObjectById(savedObjectId, SavedObjectType.ML_JOB, true);
+      await this.deleteSavedObjectById(savedObjectId, SavedObjectType.ML_JOB, true, space);
     },
 
-    async cleanMLSavedObjects() {
+    async cleanMLSavedObjects(additionalSpaces: string[] = []) {
+      // clean default space
       await this.cleanMLJobSavedObjects();
       await this.cleanMLTrainedModelsSavedObjects();
+
+      for (const space of additionalSpaces) {
+        await this.cleanMLJobSavedObjects(space);
+        await this.cleanMLTrainedModelsSavedObjects(space);
+      }
     },
 
-    async cleanMLJobSavedObjects() {
+    async cleanMLJobSavedObjects(space?: string) {
```

Comment:
The `deleteSavedObjectById` call in this method should pass through the `space`.